phase2/src/verifier_groth.sol

```diff
@@ -194,17 +194,17 @@ contract Verifier {
         Pairing.G1Point memory vk_x = Pairing.G1Point(0, 0);
 
         // Make sure that proof.A, B, and C are each less than the prime q
-        require(proof.A.X < PRIME_Q, "verifier-aX-gte-prime-q");
-        require(proof.A.Y < PRIME_Q, "verifier-aY-gte-prime-q");
+        require(_proof.A.X < PRIME_Q, "verifier-aX-gte-prime-q");
+        require(_proof.A.Y < PRIME_Q, "verifier-aY-gte-prime-q");
 
-        require(proof.B.X[0] < PRIME_Q, "verifier-cX0-gte-prime-q");
-        require(proof.B.Y[0] < PRIME_Q, "verifier-cY0-gte-prime-q");
+        require(_proof.B.X[0] < PRIME_Q, "verifier-cX0-gte-prime-q");
+        require(_proof.B.Y[0] < PRIME_Q, "verifier-cY0-gte-prime-q");
 
-        require(proof.B.X[1] < PRIME_Q, "verifier-cX1-gte-prime-q");
-        require(proof.B.Y[1] < PRIME_Q, "verifier-cY1-gte-prime-q");
+        require(_proof.B.X[1] < PRIME_Q, "verifier-cX1-gte-prime-q");
+        require(_proof.B.Y[1] < PRIME_Q, "verifier-cY1-gte-prime-q");
 
-        require(proof.C.X < PRIME_Q, "verifier-cX-gte-prime-q");
-        require(proof.C.Y < PRIME_Q, "verifier-cY-gte-prime-q");
+        require(_proof.C.X < PRIME_Q, "verifier-cX-gte-prime-q");
+        require(_proof.C.Y < PRIME_Q, "verifier-cY-gte-prime-q");
```

Comment:
since now we have the proof as `uint[8]` array above, we can iterate over it instead of adding a require() for each item explicitly. Will not be able to differentiate which exact item failed the check but I don't think we really need to.

Reply:
I agree, thank you! I'll make this change.